java-checks/src/main/java/org/sonar/java/checks/EnumMapCheck.java

```diff
@@ -38,6 +42,7 @@
 @Rule(key = "S1640")
 public class EnumMapCheck extends BaseTreeVisitor implements JavaFileScanner {
   private JavaFileScannerContext context;
+  private static final MethodMatcher mapPut = MethodMatcher.create().typeDefinition("java.util.Map").name("put").withAnyParameters();
 
   @Override
   public void scanFile(final JavaFileScannerContext context) {
```

Comment:
Not related to your change, but if this is relying on semantic, I think we should quit when semantic model is not available